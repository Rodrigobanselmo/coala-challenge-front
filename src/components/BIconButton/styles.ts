import { IconButton, styled } from "@mui/material";

export const StyledBIconButton = styled(IconButton)<{
  bg?: string;
  disabled?: boolean;
}>`
  ${({ bg }) =>
    bg &&
    `
    background-color: ${bg};
    box-shadow: rgb(0 0 0 / 5%) 0px 3px 12px;

    &:hover {
      background-color: ${bg};
      filter: brightness(0.9);
    }

    &:active {
      background-color: ${bg};
    }
  `}

  ${({ disabled }) => disabled && `opacity: 0.4;`}
`;
