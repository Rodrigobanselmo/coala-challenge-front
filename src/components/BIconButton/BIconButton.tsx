import { FC, ReactNode } from "react";

import CircularProgress from "@mui/material/CircularProgress";
import { StyledBIconButton } from "./styles";
import { SIconButtonProps } from "./types";

export const BIconButton: FC<{ children?: ReactNode } & SIconButtonProps> = ({
  circularProps,
  loading,
  children,
  disabled,
  size,
  bg,
  ...props
}) => (
  <StyledBIconButton
    bg={bg}
    disabled={loading || disabled}
    size={size}
    {...props}
  >
    {loading ? (
      <CircularProgress size={size === "small" ? 20 : 22} {...circularProps} />
    ) : (
      children
    )}
  </StyledBIconButton>
);
