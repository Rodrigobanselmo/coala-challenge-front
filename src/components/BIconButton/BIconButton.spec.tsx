import React from "react";
import { render } from "@testing-library/react";
import { BIconButton } from "./BIconButton";

describe("BIconButton", () => {
  it("matches the snapshot", () => {
    const props = {
      children: "Button Label",
    };

    const { container } = render(<BIconButton {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("matches the snapshot when loading", () => {
    const props = {
      loading: true,
      children: "Button Label 2",
    };

    const { container } = render(<BIconButton {...props} />);
    expect(container).toMatchSnapshot();
  });
});
