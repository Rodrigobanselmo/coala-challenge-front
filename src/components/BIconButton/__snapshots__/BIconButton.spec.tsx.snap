// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BIconButton matches the snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1s9kc11-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    Button Label
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`BIconButton matches the snapshot when loading 1`] = `
<div>
  <button
    class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-1xw6id3-MuiButtonBase-root-MuiIconButton-root"
    disabled=""
    tabindex="-1"
    type="button"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style="width: 22px; height: 22px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </button>
</div>
`;
